rest/views.py

```diff
@@ -0,0 +1,157 @@
+"""
+Controllers for REST app
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.generics import (
+    ListCreateAPIView,
+    RetrieveUpdateAPIView,
+    RetrieveAPIView,
+)
+from rest_framework.reverse import reverse
+from rest_framework.permissions import IsAuthenticated
+
+from roles.permissions import GroupTypes
+from roles.api import assign_user_to_repo_group
+from rest.serializers import (
+    VocabularySerializer,
+    RepositorySerializer,
+    TermSerializer,
+)
+from rest.permissions import (
+    AddRepoPermission,
+    ViewRepoPermission,
+)
+from learningresources.models import Repository
+from learningresources.api import (
+    get_repos,
+)
+
+
+# pylint: disable=too-many-ancestors
+class RepositoryList(ListCreateAPIView):
+    """REST list view for Repository"""
+    serializer_class = RepositorySerializer
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'repo_slug'
+    permission_classes = (AddRepoPermission, IsAuthenticated)
+
+    def get_success_headers(self, data):
+        """Add Location header for model create"""
+        url = reverse('repository-detail', kwargs={'repo_slug': data['slug']})
+        return {'Location': url}
+
+    def get_queryset(self):
+        """Filter to these repositories"""
+
+        repo_ids = [repo.id for repo in get_repos(self.request.user.id)]
+        return Repository.objects.filter(id__in=repo_ids).order_by('id')
+
+    def perform_create(self, serializer):
+        """Create a new repository"""
+        # pylint: disable=protected-access
+        repo = serializer.save()
+        assign_user_to_repo_group(
+            self.request.user,
+            repo,
+            GroupTypes.REPO_ADMINISTRATOR
+        )
+
+
+class RepositoryDetail(RetrieveAPIView):
+    """REST detail view for Repository"""
+    serializer_class = RepositorySerializer
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'repo_slug'
+    permission_classes = (ViewRepoPermission, IsAuthenticated)
+
+    def get_queryset(self):
+        """Filter to this repository"""
+        return Repository.objects.filter(slug=self.kwargs['repo_slug'])
```

Comment:
If this is only even meant to return a single repository, should it return a single value (using `.get`) instead of a list?